pycode/memilio-epidata/memilio/epidata/getVaccinationData.py

```diff
@@ -670,31 +712,30 @@ def get_vaccination_data(read_data=dd.defaultDict['read_data'],
                  dd.EngEng["date"]])
 
     if make_plot:
-        # have a look extrapolated vaccination ratios (TODO: to be removed or outsourced with plotting for production)
+        # have a look extrapolated vaccination ratios (TODO: create plotting for production)
         # aggregate total number of vaccinations per county and age group
         latest_date = df_data_agevacc_county_cs[dd.EngEng["date"]][len(
             df_data_agevacc_county_cs.index)-1].strftime("%Y-%m-%d")
         vacc_sums_nonsanit = df_data_agevacc_county_cs.loc[(
             df_data_agevacc_county_cs.Date == latest_date), ['ID_County', column_names_new[1]]]
         # create new data frame and reshape it
-        df_fullsum = pd.DataFrame(
-            columns=[dd.EngEng['idCounty']] + unique_age_groups_old)
-        df_fullsum[dd.EngEng['idCounty']
-                   ] = vacc_sums_nonsanit[dd.EngEng['idCounty']].unique()
-        df_fullsum[unique_age_groups_old] = np.array(
-            vacc_sums_nonsanit[column_names_new[1]]).reshape(
-            len(vacc_sums_nonsanit[dd.EngEng['idCounty']].unique()),
-            len(unique_age_groups_old))
-        # compute county and age-group-specific vaccination ratios
-        df_fullsum[['r'+age for age in unique_age_groups_old]] = df_fullsum[unique_age_groups_old] / geoger.merge_df_counties_all(
-            population_old_ages, sorting=[dd.EngEng["idCounty"]], columns=dd.EngEng["idCounty"])[unique_age_groups_old].values
-        ## end of the be removed ###
-
-    # store data
+        df_fullsum = compute_vaccination_ratios(unique_age_groups_old, vacc_sums_nonsanit, column_names_new[1], dd.EngEng['idCounty'], population_old_ages, merge_2022=True)
+
+    # store data for all counties
     filename = 'all_county_agevacc_vacc'
     filename = gd.append_filename(filename, impute_dates, moving_average)
     gd.write_dataframe(df_data_agevacc_county_cs,
                        directory, filename, file_format)
+
+    # store data for all federal states: group information on date, state and age level 
+    # (i.e., aggregate information of all counties per federal state)
+    filename = 'all_states_agevacc_vacc'
+    filename = gd.append_filename(filename, impute_dates, moving_average)
+    df_data_agevacc_state_cs = df_data_agevacc_county_cs.groupby(
```

Comment:
This dataframe is just created to be written in the pydata folder. Do we need this in here?

Reply:
Not here, but in cpp code this can be read in.